### PR TITLE
Implement exclusive checkboxes

### PR DESCRIPTION
**Pull request by kosiakkatrina**
Set all the checkboxes after a divider to be exclusive

### CODE REVIEW

File: app/views/form/_checkbox_question.html.erb [hunk 1]
```diff
@@ -4,14 +4,17 @@
     caption: caption(caption_text, page_header, conditional),
     legend: legend(question, page_header, conditional),
     hint: { text: question.hint_text&.html_safe } do %>
+    <% after_divider = false %>
 
     <% question.answer_options.map do |key, val| %>
       <% if key.starts_with?("divider") %>
-        <%= f.govuk_check_box_divider %>
+      <% after_divider = true %>
```
Comment (baarkerlounger) [inline app/views/form/_checkbox_question.html.erb:11]: Interesting approach! Is it definitely always the case that anything separated by a divider is mutually exclusive? @paulrobertlloyd? If so, then :very_nice:

Comment (kosiakkatrina) [reply app/views/form/_checkbox_question.html.erb:11]: Hmm, well our dividers put "or" between the options, so I'd assume so 👀

**Comment by baarkerlounger** [comment]
Think we also need to add backend validations for this for when JS is turned off and/or API submission?

**Comment by kosiakkatrina** [comment]
> Think we also need to add backend validations for this for when JS is turned off and/or API submission?

@baarkerlounger Yeah, I think we've got those defined in separate tickets. One of them is already implemented, but thanks for bringing it up, cause it actually needed fixing (done now)

File: app/views/form/_checkbox_question.html.erb (hunk 1 above)
Comment (paulrobertlloyd) [reply app/views/form/_checkbox_question.html.erb:11]: Yup, we’d want to separate out those options that are mutually exclusive after a divider. See https://design-system.service.gov.uk/components/checkboxes/#add-an-option-for-none-

**Review by baarkerlounger — APPROVED** [review]

File: app/views/form/_checkbox_question.html.erb (hunk 1 above)
Comment (paulrobertlloyd) [reply app/views/form/_checkbox_question.html.erb:11]: Whether we’d use a divider for other use cases with checkboxes I don’t know, but I suspect not. This is a good assumption to go forward with, for now.

**Pull request merged**